File: src/main/java/ting/leetcode/java/arrays101/FindTheTwoJudge.java
package ting.leetcode.java.arrays101;

public class FindTheTwoJudge {
    public int findJudge(int n, int[][] trust) {
        int[] trustPoint = new int[n + 1];

        for (int[] t : trust) {
            trustPoint[t[0]]--;
            trustPoint[t[1]]++;
        }
        for (int j = 1; j < trustPoint.length; j++) {
            if (trustPoint[j] == n - 1)
                return j;
        }
        return -1;
    }
}

File: kotlin-solution/src/leetcode/arrays101/FindTheTwoJudge.kt
package leetcode.arrays101

class FindTheTwoJudge {
    fun findJudge(n: Int, trust: Array<IntArray>): Int {
        val trustPoint = IntArray(n + 1)
        for (relation in trust) {
            trustPoint[relation.first()]--
            trustPoint[relation.last()]++
        }

        for (i in 1..trustPoint.lastIndex) {
            if (trustPoint[i] == n - 1)
                return i
        }
        return -1
    }
}
